cmd/openstack-populator/openstack-populator.go

```diff
@@ -93,38 +97,54 @@ func populate(fileName, identityEndpoint, secretName, imageID string) {
 	}
 	defer file.Close()
 
-	err = writeData(imageReader, file, imageID, progressGague)
+	err = writeData(imageReader, file, imageID, ownerUID, progressVec, &pvcSize)
 	if err != nil {
 		klog.Fatal(err)
 	}
 }
 
 type CountingReader struct {
 	reader io.ReadCloser
-	total  *int64
+	read   *int64
 }
 
 func (cr *CountingReader) Read(p []byte) (int, error) {
 	n, err := cr.reader.Read(p)
-	*cr.total += int64(n)
+	*cr.read += int64(n)
 	return n, err
 }
 
-func writeData(reader io.ReadCloser, file *os.File, imageID string, progress *prometheus.GaugeVec) error {
-	total := new(int64)
+func writeData(reader io.ReadCloser, file *os.File, imageID, ownerUID string, progress *prometheus.CounterVec, total *int64) error {
 	countingReader := CountingReader{reader, total}
 
 	done := make(chan bool)
+
 	go func() {
+		metric := &dto.Metric{}
+		var currentProgress float64
+
 		for {
 			select {
 			case <-done:
+				if metric.Counter != nil {
+					progress.WithLabelValues(ownerUID).Add(100 - *metric.Counter.Value)
```

Comment:
why don't we write `100` here like in the ovirt populator below?

Reply:
I'm not sure it works yet, I don't see this reported in neither cases